docs/content/envoy-tutorial-gloo-edge.md

```diff
@@ -0,0 +1,314 @@
+
+The purpose of this tutorial is to show how OPA could be used with Gloo Edge to apply security policies for upstream services.
+
+## Prerequisites
+
+This tutorial requires Kubernetes 1.14 or later. To run the tutorial locally, we recommend using [minikube](https://minikube.sigs.k8s.io/docs/start/) in version v1.0+ with Kubernetes 1.14+.
+
+The tutorial also requires [Helm](https://helm.sh/docs/intro/install/) to install Gloo Edge on a Kubernetes cluster.
+
+## Steps
+
+### 1. Start Minikube
+
+```bash
+minikube start
+```
+
+### 2. Setup and Configure Gloo Edge
+
+```bash
+helm repo add gloo https://storage.googleapis.com/solo-public-helm
+helm upgrade --install --namespace gloo-system --create-namespace gloo gloo/gloo
+kubectl config set-context $(kubectl config current-context) --namespace=gloo-system
+```
+
+Ensure all the pods are running using `kubectl get pod` command.
+
+### 3. Create Virtual Service and Upstream
+
+[Virtual Services](https://docs.solo.io/gloo-edge/latest/introduction/architecture/concepts/#virtual-services) define a set of route rules, security configuration, rate limiting, transformations, and other core routing capabilities supported by Gloo Edge.
+
+[Upstreams](https://docs.solo.io/gloo-edge/latest/introduction/architecture/concepts/#upstreams) define destinations for routes.
+
+Save the configuration as **vs.yaml**.
+
+```yaml
+apiVersion: gloo.solo.io/v1
+kind: Upstream
+metadata:
+  name: httpbin
+spec:
+  static:
+    hosts:
+      - addr: httpbin.org
+        port: 80
+---
+apiVersion: gateway.solo.io/v1
+kind: VirtualService
+metadata:
+  name: httpbin
+spec:
+  virtualHost:
+    domains:
+      - '*'
+    routes:
+      - matchers:
+         - prefix: /
+        routeAction:
+          single:
+            upstream:
+              name: httpbin
+              namespace: gloo-system
+        options:
+          autoHostRewrite: true
+```
+
+```bash
+kubectl apply -f vs.yaml
+```
+
+### 4. Test Gloo
+
+For simplification port-forwarding will be used. Open another terminal and execute.
+
+```bash
+kubectl port-forward deployment/gateway-proxy 8080:8080
+```
+
+The `VirtualService` created in the previous step forwards requests to http://httpbin.org
+
+Let's test that Gloo works properly by running the below command in the first terminal.
+
+```bash
+curl -XGET -Is localhost:8080/get | head -n 1
+HTTP/1.1 200 OK
+
+curl http -XPOST -Is localhost:8080/post | head -n1
+HTTP/1.1 200 OK
+```
+
+### 5. Define an OPA Policy
+
+The following OPA policy will work as follow:
+
+* Alice is granted a **guest** role and can perform `GET` requests.
+* Bob is granted an **admin** role and can perform `GET` and `POST` requests.
+
+**policy.rego**
```

Comment:
Let's update this so that the policy can be opened in the Rego Playground. See [this](https://raw.githubusercontent.com/open-policy-agent/opa/main/docs/content/envoy-tutorial-standalone-envoy.md) example on how to do this. We need to add a tag like below
```
```live:example:module:openable
```

Also let's show the input OPA gets. The above link has an example on how to make the input live.

Reply:
Applied this, quite cool, will share the netlify link for it